diff --git a/node_modules/filer.js/src/filer.js b/node_modules/filer.js/src/filer.js
index dca5cd4..2a59fd1 100755
--- a/node_modules/filer.js/src/filer.js
+++ b/node_modules/filer.js/src/filer.js
@@ -38,7 +38,7 @@ self.BlobBuilder = self.BlobBuilder || self.MozBlobBuilder ||
 
 // Prevent errors in browsers that don't support FileError.
 if (self.FileError === undefined) {
-  var FileError = function() {};
+  self.FileError = function() {};
   FileError.prototype.prototype = Error.prototype;
 }
 
