diff --git a/node_modules/filer.js/src/filer.js b/node_modules/filer.js/src/filer.js
index dca5cd4..ebd66c4 100644
--- a/node_modules/filer.js/src/filer.js
+++ b/node_modules/filer.js/src/filer.js
@@ -38,7 +38,7 @@ self.BlobBuilder = self.BlobBuilder || self.MozBlobBuilder ||
 
 // Prevent errors in browsers that don't support FileError.
 if (self.FileError === undefined) {
-  var FileError = function() {};
+  self.FileError = function() {};
   FileError.prototype.prototype = Error.prototype;
 }
 
@@ -300,7 +300,7 @@ var Filer = new function() {
     var destStr = arguments[2];
 
     var onError = function(e) {
-      if (e.code == FileError.NOT_FOUND_ERR) {
+      if (e.code == FileError.NOT_FOUND_ERR || e.name === "NotFoundError") {
         if (destStr) {
           throw new Error('"' + srcStr + '" or "' + destStr +
                           '" does not exist.');
