diff --git a/node_modules/filer.js/src/filer.js b/node_modules/filer.js/src/filer.js
index dca5cd4..eb013d7 100755
--- a/node_modules/filer.js/src/filer.js
+++ b/node_modules/filer.js/src/filer.js
@@ -38,7 +38,7 @@ self.BlobBuilder = self.BlobBuilder || self.MozBlobBuilder ||
 
 // Prevent errors in browsers that don't support FileError.
 if (self.FileError === undefined) {
-  var FileError = function() {};
+  self.FileError = function() {};
   FileError.prototype.prototype = Error.prototype;
 }
 
@@ -294,36 +294,48 @@ var Filer = new function() {
    *     to this method.
    * @param {...string} var_args 1-2 paths to lookup and return entries for.
    *     These can be paths or filesystem: URLs.
+   *
+   * @param {errorCallback} optional last parameter which can be executed if an error occurs
    */
   var getEntry_ = function(callback, var_args) {
     var srcStr = arguments[1];
-    var destStr = arguments[2];
+    var errorCallback;
+    var destStr;
+
+    if (typeof arguments[2] == "string") {
+      destStr = arguments[2];
+      errorCallback = arguments[3];
+    } else {
+      errorCallback = arguments[2];
+    }
 
-    var onError = function(e) {
-      if (e.code == FileError.NOT_FOUND_ERR) {
-        if (destStr) {
-          throw new Error('"' + srcStr + '" or "' + destStr +
-                          '" does not exist.');
+    if (!errorCallback) {
+      errorCallback = function(e) {
+        if (e.code == FileError.NOT_FOUND_ERR || e.name == "NotFoundError") {
+          if (destStr) {
+            throw new Error('"' + srcStr + '" or "' + destStr +
+                            '" does not exist.');
+          } else {
+            throw new Error('"' + srcStr + '" does not exist.');
+          }
         } else {
-          throw new Error('"' + srcStr + '" does not exist.');
+          throw new Error('Problem getting Entry for one or more paths.');
         }
-      } else {
-        throw new Error('Problem getting Entry for one or more paths.');
-      }
-    };
+      };
+    }
 
     // Build a filesystem: URL manually if we need to.
     var src = pathToFsURL_(srcStr);
 
-    if (arguments.length == 3) {
+    if ((typeof srcStr == "string") && (typeof destStr == "string")) {
       var dest = pathToFsURL_(destStr);
       self.resolveLocalFileSystemURL(src, function(srcEntry) {
         self.resolveLocalFileSystemURL(dest, function(destEntry) {
           callback(srcEntry, destEntry);
-        }, onError);
-      }, onError);
+        }, errorCallback);
+      }, errorCallback);
     } else {
-      self.resolveLocalFileSystemURL(src, callback, onError);
+      self.resolveLocalFileSystemURL(src, callback, errorCallback);
     }
   };
 
@@ -477,13 +489,15 @@ var Filer = new function() {
 
     var callback = function(dirEntry) {
 
-      cwd_ = dirEntry;
+      // Don't set cwd_ here
+      // https://github.com/ebidel/filer.js/pull/20/files
+      // cwd_ = ;
 
       // Read contents of current working directory. According to spec, need to
       // keep calling readEntries() until length of result array is 0. We're
       // guarenteed the same entry won't be returned again.
       var entries_ = [];
-      var reader = cwd_.createReader();
+      var reader = dirEntry.createReader();
 
       var readEntries = function() {
         reader.readEntries(function(results) {
@@ -595,7 +609,7 @@ var Filer = new function() {
     } else {
       getEntry_(function(fileEntry) {
         fileEntry.file(successCallback, opt_errorHandler);
-      }, pathToFsURL_(entryOrPath));
+      }, pathToFsURL_(entryOrPath), opt_errorHandler);
     }
   };
 
