diff --git a/node_modules/pixi.js/pixi.js.d.ts b/node_modules/pixi.js/pixi.js.d.ts
index f9c7247..16110df 100644
--- a/node_modules/pixi.js/pixi.js.d.ts
+++ b/node_modules/pixi.js/pixi.js.d.ts
@@ -12178,7 +12178,7 @@ declare namespace PIXI {
          * @protected
          * @member {number} PIXI.Transform#_localID
          */
-        protected _localID: number;
+        _localID: number;
         /**
          * The locally unique ID of the local transform
          * used to calculate the current local transformation matrix.
@@ -12186,14 +12186,14 @@ declare namespace PIXI {
          * @protected
          * @member {number} PIXI.Transform#_currentLocalID
          */
-        protected _currentLocalID: number;
+        _currentLocalID: number;
         /**
          * The locally unique ID of the world transform.
          *
          * @protected
          * @member {number} PIXI.Transform#_worldID
          */
-        protected _worldID: number;
+        _worldID: number;
         /**
          * The locally unique ID of the parent's world transform
          * used to calculate the current world transformation matrix.
@@ -12201,7 +12201,7 @@ declare namespace PIXI {
          * @protected
          * @member {number} PIXI.Transform#_parentID
          */
-        protected _parentID: number;
+        _parentID: number;
         /**
          * Called when a value changes.
          *
@@ -18597,7 +18597,7 @@ declare namespace PIXI {
          * @protected
          * @param {PIXI.Renderer} renderer - The webgl renderer to use.
          */
-        protected _render(renderer: PIXI.Renderer): void;
+        _render(renderer: PIXI.Renderer): void;
         /**
          * Updates the bounds of the sprite.
          *
@@ -19499,7 +19499,7 @@ declare namespace PIXI {
          * @protected
          * @param {PIXI.Renderer} renderer - The webgl renderer to use.
          */
-        protected _render(renderer: PIXI.Renderer): void;
+        _render(renderer: PIXI.Renderer): void;
         /**
          * Updates the bounds of the sprite.
          *
@@ -20214,7 +20214,7 @@ declare namespace PIXI {
          * @protected
          * @param {PIXI.Renderer} renderer - The renderer
          */
-        protected _render(renderer: PIXI.Renderer): void;
+        _render(renderer: PIXI.Renderer): void;
         /**
          * Updates the bounds of the tiling sprite.
          *
