diff --git a/node_modules/pixi-keyboard/package.json b/node_modules/pixi-keyboard/package.json
index c8e8927..0d67213 100644
--- a/node_modules/pixi-keyboard/package.json
+++ b/node_modules/pixi-keyboard/package.json
@@ -2,7 +2,7 @@
   "name": "pixi-keyboard",
   "version": "0.9.4",
   "description": "pixi-keyboard is a plugin for Pixi.js v3.0.8 or higher to manage the keyboard events easily.",
-  "main": "./build/pixi-keyboard.js",
+  "main": "./src/index.js",
   "author": "Nazariglez <nazari.nz@gmail.com> (http://www.nazariglez.com)",
   "license": "MIT",
   "keywords": [
diff --git a/node_modules/pixi-keyboard/src/KeyboardManager.js b/node_modules/pixi-keyboard/src/KeyboardManager.js
index 087b558..0af0124 100644
--- a/node_modules/pixi-keyboard/src/KeyboardManager.js
+++ b/node_modules/pixi-keyboard/src/KeyboardManager.js
@@ -1,7 +1,7 @@
-import PIXI from 'pixi.js';
+import { utils } from 'pixi.js';
 import HotKey from './HotKey';
 
-export default class KeyboardManager extends PIXI.utils.EventEmitter {
+export default class KeyboardManager extends utils.EventEmitter {
   constructor(){
     super();
     this.isEnabled = false;
@@ -76,6 +76,19 @@ export default class KeyboardManager extends PIXI.utils.EventEmitter {
     }
   }
 
+  onKeyPressedWithPreventDefault(key, callback) {
+    this.onKeysPressedWithPreventDefault([key], callback);
+  }
+
+  onKeysPressedWithPreventDefault(keys, callback) {
+    this.setPreventDefault(keys);
+    this.on("pressed", () => {
+      if (keys.every(key => this.isPressed(key))) {
+        callback();
+      }
+    })
+  }
+
   isDown(key){
     return (this._downKeys.indexOf(key) !== -1);
   }
diff --git a/node_modules/pixi-keyboard/src/index.js b/node_modules/pixi-keyboard/src/index.js
index 288e509..2af2366 100644
--- a/node_modules/pixi-keyboard/src/index.js
+++ b/node_modules/pixi-keyboard/src/index.js
@@ -1,19 +1,7 @@
-import PIXI from'pixi.js';
-import KeyboardManager from './KeyboardManager';
-import HotKey from './HotKey';
-import Key from './Key';
+import _KeyboardManager from './KeyboardManager';
+import _HotKey from './HotKey';
+import _Key from './Key';
 
-let keyboard = {
-  KeyboardManager: KeyboardManager,
-  Key: Key,
-  HotKey: HotKey
-};
-
-if(!PIXI.keyboard){
-  let keyboardManager = new KeyboardManager();
-  keyboardManager.enable();
-
-  PIXI.keyboard = keyboard;
-  PIXI.keyboardManager = keyboardManager;
-}
-export default keyboard;
+export const KeyboardManager = _KeyboardManager;
+export const Key = _Key;
+export const HotKey = _HotKey;
